packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -183,25 +210,69 @@ export default function VisualEditor( { styles } ) {
 
 	const blockSelectionClearerRef = useBlockSelectionClearer();
 
-	const layout = useMemo( () => {
+	// fallbackLayout is used if there is no Post Content,
+	// and for Post Title.
+	const fallbackLayout = useMemo( () => {
 		if ( isTemplateMode ) {
 			return { type: 'default' };
 		}
 
 		if ( themeSupportsLayout ) {
 			// We need to ensure support for wide and full alignments,
 			// so we add the constrained type.
-			return { ...defaultLayout, type: 'constrained' };
+			return { ...globalLayoutSettings, type: 'constrained' };
 		}
 		// Set default layout for classic themes so all alignments are supported.
 		return { type: 'default' };
-	}, [ isTemplateMode, themeSupportsLayout, defaultLayout ] );
+	}, [ isTemplateMode, themeSupportsLayout, globalLayoutSettings ] );
+
+	const postContentBlock = useMemo( () => {
+		// When in template editing mode, we can access the blocks directly.
+		if ( editedPostTemplate?.blocks ) {
+			return findPostContent( editedPostTemplate?.blocks );
```

Comment:
I can reproduce error that @andrewserong  [noticed](https://github.com/WordPress/gutenberg/pull/44258#pullrequestreview-1119518999) but only sometimes. 

Most of the times it works for me. 

When it does fail _I suspect_ it's because postContentBlock is undefined (and therefore trying to access the attributes below fails).

Replacing with `return findPostContent( editedPostTemplate?.blocks ) || {};` seems to get around that, 

![2022-09-26 12 39 39](https://user-images.githubusercontent.com/6458278/192183452-1d765f0a-9f84-48f2-be7d-77e848d27384.gif)

But it exposes another error triggered by the <BlockPopoverInbetween> component 🤣 

`Uncaught TypeError: Cannot read properties of null (reading 'removeEventListener')`

https://github.com/WordPress/gutenberg/blob/ed8d25e651f9673583ae1c4800339eec45684830/packages/block-editor/src/components/block-popover/inbetween.js#L229

Not sure why yet. I can't reproduce either error in trunk.

Reply:
Hmm, I've only managed to repro the error once on TT2, and not at all on Empty theme. But I think something's messed up because previous customisations are now appearing as the default template state 😅 
Will continue investigating.

Reply:
Ok I fixed the failure if Post Content block is removed from the template by accessing layout from attributes via optional chaining instead of destructuring. The reason returning empty object from `findPostContent` won't work correctly is some logic further down is dependent on checking if postContentBlock exists, and empty object will always be `true` 😅